scripts/publish.js

```diff
@@ -46,6 +47,28 @@ if (!publishers.includes('salesforce')) {
   exit(-1);
 }
 
+// Checks that you have specified the next version as an environment variable, and that it's properly formatted.
+const nextVersion = process.env['SALESFORCEDX_VSCODE_VERSION'];
+if (!nextVersion) {
+  console.log(
+    'You must specify the next version of the extension by setting SALESFORCEDX_VSCODE_VERSION as an environment variable.'
```

Comment:
Put at the top of this file to run with `SALESFORCE_VSCODE_VERSION=x.y.z ./scripts/publish.js`

That should be a good example.